File: src/core/DiceRollSim.java
package core;

import java.util.Random;

public class DiceRollSim {
    
    private boolean running = true;
    
    


    public void diceRollCore() {
        Random random = new Random();
        
        while(getRunning()) { // Main entry in loop





        }







        
        
        System.out.println("Do you want to roll some dice? (y/n)");
        char diceRoll = scanner.next().charAt(0);
        
        boolean running = true;
        boolean repeat = false;
        int numberOfDiceRolls = 0;
        int time1 = 0;
        int time2 = 0;
        int time3 = 0;
        int time4 = 0;
        int time5 = 0;
        int time6 = 0;

        final double ONETOSIX = 0.1666;
        while (running) {
            if (diceRoll == 'y' || diceRoll == 'Y') {

                double diceAccumulation = 0;
                if (!(repeat)) {
                    System.out.println("How many dice do you want to roll?");
                    numberOfDiceRolls = scanner.nextInt();
                }

                System.out.println("Here are your dice rolls:");
                for (int i = 0; i <= numberOfDiceRolls - 1; i++) {
                    int randomDiceRoll = random.nextInt(1, 7);

                    switch (randomDiceRoll) {
                        case 1:
                            time1++;
                            break;
                        case 2:
                            time2++;
                            break;
                        case 3:
                            time3++;
                            break;
                        case 4:
                            time4++;
                            break;
                        case 5:
                            time5++;
                            break;
                        case 6:
                            time6++;
                            break; 
                    }
                    diceAccumulation += randomDiceRoll;
                    System.out.print(randomDiceRoll);
                    if (numberOfDiceRolls > 1 && (i != numberOfDiceRolls - 1)) {
                        System.out.print(", ");
                    }
                }

                
                double diceAverage = ((double) (diceAccumulation / numberOfDiceRolls));
                double diceDeviation = (3.5 - diceAverage);
                diceDeviation = Math.abs(diceDeviation);
                
                double portion1 = (double) time1 / numberOfDiceRolls;
                double portion2 = (double) time2 / numberOfDiceRolls;
                double portion3 = (double) time3 / numberOfDiceRolls;
                double portion4 = (double) time4 / numberOfDiceRolls;
                double portion5 = (double) time5 / numberOfDiceRolls;
                double portion6 = (double) time6 / numberOfDiceRolls ;

                System.out.println();
                System.out.println("Just as a reminder, you rolled " + numberOfDiceRolls + " dice.");
                System.out.println("This is how often you rolled a '1': " + time1 + ". That equals " + (portion1 * 100) + " %.");
                System.out.println("This is how often you rolled a '2': " + time2 + ". That equals " + (portion2 * 100) + " %.");
                System.out.println("This is how often you rolled a '3': " + time3 + ". That equals " + (portion3 * 100) + " %.");
                System.out.println("This is how often you rolled a '4': " + time4 + ". That equals " + (portion4 * 100) + " %.");
                System.out.println("This is how often you rolled a '5': " + time5 + ". That equals " + (portion5 * 100) + " %.");
                System.out.println("This is how often you rolled a '6': " + time6 + ". That equals " + (portion6 * 100) + " %.");
                System.out.println("And to remind you of another fact, 1/6 equals " + ONETOSIX*100 + " %.");
                System.out.println("Your last dice roll average equaled to: " + diceAverage);
                System.out.println("The deviation of the theoretical average point of 3.5 equals to: " + diceDeviation);
                System.out.println("The deviation times one hundred equals to: " + diceDeviation * 100);
                System.out.println("So anyway, ...");
                System.out.println("Do you want to roll the dice once again or repeat what you just did? (y/n/r)");
                char anotherRoll = scanner.next().charAt(0);
                if (anotherRoll == 'y' || anotherRoll == 'Y') {
                    repeat = false;
                } else if (anotherRoll == 'r' || anotherRoll == 'R') {
                    repeat = true;
                } else {
                    System.out.println("You failed!");
                    running = false;
                }
                time1 = 0;
                time2 = 0;
                time3 = 0;
                time4 = 0;
                time5 = 0;
                time6 = 0;
            } else {
                System.out.println("You failed!");
                running = false;
            }
        }
    }

    public boolean getRunning() {
        return running;
    }

    public void setRunning(boolean running) {
        this.running = running;
    }

    public boolean startDiceRoll() {
        
        return false;
    }

    
}

File: dice_roll_sim.py
import random
import time

def get_user_input(prompt, valid_responses):
    while True:
        user_input = input(prompt)
        if user_input.upper() in valid_responses:
            return user_input.upper()
        else:
            print("Please answer correctly.")

def get_number_of_dice():
    while True:
        user_input = input("How many dice do you want to roll? ")
        if user_input.isdigit() and int(user_input) > 0:
            return int(user_input)
        else:
            print("Again, please answer correctly with a number.")

def dice_gen(number_of_dice):
    dice_rolls = []
    for _ in range(1, number_of_dice + 1):
        roll = random.randint(1, 6)
        dice_rolls.append(roll)
        print(roll, end='', flush=True)
        if _ != number_of_dice:
            print(", ", end='')
    return dice_rolls

def statistics(number_of_dice, dice_rolls):
    dice_amount = [0] * 6
    portion = [0] * 6
    average = sum(dice_rolls) / number_of_dice
    deviation = abs(3.5 - average)
    for dice in dice_rolls:
        dice_amount[dice - 1] += 1
    for i in range(6):
        portion[i] = dice_amount[i] / number_of_dice
    return [number_of_dice, dice_amount, portion, average, deviation]

def printer(stats):
    print(f"\b\nJust as a reminder, you rolled {stats[0]} dice.")
    for index in range(6):
        print(f"This is how often you rolled a '{index + 1}': {stats[1][index]}. "
              f"That equals {stats[2][index] * 100} %.")
    print(f"And to remind you of another fact, 1/6 equals 16.66 %.")
    print(f"Your last dice roll average equaled to: {stats[3]}")
    print(f"The deviation of the theoretical average point of 3.5 equals to: {stats[4]}")
    print(f"The deviation times one hundred equals to: {stats[4] * 100}")
    print(f"So anyway, ...")

def main():
    running = get_user_input("Do you want to roll some dice? (y/n): ", ['Y', 'N'])
    number_of_dice = 0
    while running == 'Y' or running == 'R':
        if running != 'R':
            number_of_dice = get_number_of_dice()
        print("Here are your dice rolls:")
        dice_rolls = dice_gen(number_of_dice)
        stats = statistics(number_of_dice, dice_rolls)
        printer(stats)
        running = get_user_input("Do you want to roll the dice once again or repeat? (y/n/r): ", ['Y', 'N', 'R'])
    print("You exited.")
    time.sleep(1)

if __name__ == "__main__":
    main()
